### PR TITLE
Add api docs

### PR DESCRIPTION
**Pull request by vmx**
The JSDocs for the callbacks are currently not proper JSDoc comments
as it seems that ESLint doesn't fully support the @callback syntax.

### CODE REVIEW

**Comment by vmx** [comment]
@diasdavid I'd especially like a review of the README change. I'm not sure if I got the wording right (in regards to "resolver","format spec" etc.).

File: src/resolver.js [hunk 1]
```diff
@@ -2,6 +2,26 @@
...
+ * Resolves a path in a Bitcoin block.
+ *
+ * Returns the value or a link and the partial mising path. This way the
+ * IPLD Resolver can fetch the link and continue to resolve.
+ *
+ * @param {Buffer} binaryBlob - Binary representation of a Bitcoin block
+ * @param {string} [path='/'] - Path that should be resolved
+ * @param {ResolveCallback} callback - Callback that handles the return value
+ */
+/*
```
Comment (richardschneider) [inline src/resolver.js]: Shouldn't this be `/**`?

File: src/resolver.js [hunk 2]
```diff
@@ -49,6 +69,23 @@ const resolve = (binaryBlob, path, callback) => {
...
+/**
+ * Return all available paths of a block.
+ *
+ * @param {Buffer} binaryBlob - Binary representation of a Bitcoin block
+ * @param {Object} [options] - Possible options
+ * @param {boolean} [options.values=false] - Retun only the paths by default.
+ *   If it is `true` also return the values
+ * @param {TreeCallback} callback - Callback that handles the return value
+ */
+/*
```
Comment (richardschneider) [inline src/resolver.js]: And again, missing double star.

File: src/util.js [hunk 3]
```diff
@@ -5,18 +5,58 @@ const CID = require('cids')
...
 const sha256 = require('hash.js/lib/hash/sha/256')
 
+/**
+ * Serialize internal representation into a binary Bitcoin block.
+ *
+ * @param {Buffer} binaryBlob - Binary representation of a Bitcoin block
+ * @param {SerializeCallback} callback - Callback that handles the
+ *   return value
+ */
+/*
```
Comment (richardschneider) [inline src/util.js]: And again, missing double star.

File: src/util.js [hunk 4]
```diff
@@ -5,18 +5,58 @@ const CID = require('cids')
...
   callback(err, dagNode)
 }
 
+/**
+ * Get the CID of the DAG-Node.
+ *
+ * @param {BitcoinBlock} dagNode - Internal representation of a Bitcoin block
+ * @param {CidCallback} callback - Callback that handles the return value
+ */
+/*
```
Comment (richardschneider) [inline src/util.js]: And again, missing double star.

**Comment by vmx** [comment]
@richardschneider ESLint doesn't seem to support `@callback` properly, hence the comments are without `/**` for now.

**Pull request merged**